### PR TITLE
✨ New user role: product-owner  🗃️

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

A new user role for product-owners shall give exclusive access to product-related entry-points

- [x] ✨ new `PRODUCT_OWNER` role (between tester and admin) 🗃️
- [x] Add new role in the `access_role` registry  (Adds a tag that is currently unused). 
- NEXT PR will give access to this new role to create invitations and to get detailed info about any product

## Related issue/s

- Implementing PO roles for https://github.com/ITISFoundation/osparc-simcore/issues/4752

## How to test

Driven by packages/postgres-database/tests/test_users.py

## DevOps

- Check migration of `users` table  🗃️

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: services/api-server/src/simcore_service_api_server/models/schemas/profiles.py [hunk 1]
```diff
@@ -17,10 +18,12 @@ class ProfileUpdate(ProfileCommon):
 
 
 class UserRoleEnum(StrAutoEnum):
+    # NOTE: this is in sync with simcore_postgres_database.models.users.UserRole via testing
     ANONYMOUS = auto()
     GUEST = auto()
     USER = auto()
     TESTER = auto()
+    PRODUCT_OWNER = auto()
```
Comment (GitHK) [inline services/api-server/src/simcore_service_api_server/models/schemas/profiles.py:26]: Q: what impact will this have on things like garbage collection and permissions. Have you thought about those?

**Review by GitHK — APPROVED** [review]
👍

File: services/api-server/src/simcore_service_api_server/models/schemas/profiles.py (hunk 1 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/models/schemas/profiles.py:26]: yes

**Pull request merged**